eclair-core/src/main/scala/fr/acinq/eclair/payment/send/MultiPartPaymentLifecycle.scala

```diff
@@ -193,6 +173,11 @@ class MultiPartPaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig,
       require(ps.parts.length == 1, "child payment must contain only one part")
       log.warning(s"payment recipient fulfilled incomplete multi-part payment (id=${ps.parts.head.id})")
       goto(PAYMENT_SUCCEEDED) using PaymentSucceeded(d.sender, d.request, ps.paymentPreimage, ps.parts, d.pending - ps.parts.head.id)
+
+    case Event(Stop, d: PaymentAborted) => myStop(d.sender, Left(PaymentFailed(id, paymentHash, d.failures)))
+
+    case Event(_: RouteResponse, _) => stay
+    case Event(_: Status.Failure, _) => stay
```

Comment:
This should never happen right? Feels like the default behavior of displaying unhandled messages as `WARN` would be better?

Reply:
This can (and will) happen in normal scenario, this is why I didn't log it.
If you receive an error from an intermediate node and ask the router for a new set of routes, and then receive an error from the final recipient, we will go to `PAYMENT_ABORTED` (because it's an error from the final recipient) while we're still waiting for a route response.
Once we're in that state these routes don't matter anymore, so we just handle the message and ignore it.